src/sentry/static/sentry/app/views/settings/projectAlerts/issueEditor/mailActionFields.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import SelectMembers from 'app/components/selectMembers';
+import SelectControl from 'app/components/forms/selectControl';
+import {Organization, Project} from 'app/types';
+import {IssueAlertRuleAction, MailActionTargetType} from 'app/types/alerts';
+import {PanelItem} from 'app/components/panels';
+import space from 'app/styles/space';
+
+type Props = {
+  project: Project;
+  organization: Organization;
+  disabled: boolean;
+  loading: boolean;
+  action: IssueAlertRuleAction;
+  onChange: (action: IssueAlertRuleAction) => void;
+};
+
+interface OptionRecord {
+  value: string;
+  label: string;
+}
+
+class MailActionFields extends React.Component<Props> {
+  handleChange = (
+    attribute: 'targetType' | 'targetIdentifier',
+    e: OptionRecord & {[key: string]: any}
+  ) => {
+    const {onChange, action} = this.props;
+    if (e.value === action[attribute]) {
+      return;
+    }
+    const newAction = {
+      ...action,
+      [attribute]: `${e.value}`,
+    };
+    /**
+     * TargetIdentifiers between the targetTypes are not unique, and may wrongly map to something that has not been
+     * selected. E.g. A member and project can both have the `targetIdentifier`, `'2'`. Hence we clear the identifier.
+     **/
+    if (attribute === 'targetType') {
+      newAction.targetIdentifier = '';
+    }
+    onChange(newAction);
+  };
+
+  handleChangeActorType = (e: OptionRecord) => {
+    this.handleChange('targetType', e);
+  };
+
+  handleChangeActorId = (e: OptionRecord & {[key: string]: any}) => {
+    this.handleChange('targetIdentifier', e);
+  };
```

Comment:
Same here, let's rename `e` as we generally use that to mean `event`.

Reply:
Done, thanks!